terminal.go

```diff
@@ -850,22 +855,33 @@ func (g *LightningTerminal) startInternalSubServers(
 		return nil
 	}
 
+	closeAccountService := func() {
+		if err := g.accountService.Stop(); err != nil {
+			// We only log the error if we fail to stop the service,
+			// as it's not critical that this succeeds in order to
+			// keep litd running
+			log.Errorf("Error stopping account service: %v", err)
+		}
+	}
+
 	log.Infof("Starting LiT account service")
-	err = g.accountService.Start(
-		g.lndClient.Client, g.lndClient.Router,
-		g.lndClient.ChainParams,
-	)
-	if err != nil {
-		log.Errorf("error starting account service: %v, disabling "+
-			"account service", err)
-		g.statusMgr.SetErrored(subservers.ACCOUNTS, err.Error())
+	if g.cfg.AccountsMode != ModeDisable {
+		err = g.accountService.Start(
+			g.lndClient.Client, g.lndClient.Router,
+			g.lndClient.ChainParams,
+		)
+		if err != nil {
+			log.Errorf("error starting account service: %v, "+
+				"disabling account service", err)
+			g.statusMgr.SetErrored(subservers.ACCOUNTS, err.Error())
+			closeAccountService()
+		} else {
+			g.statusMgr.SetRunning(subservers.ACCOUNTS)
+			g.accountServiceStarted = true
+		}
 	} else {
-		g.statusMgr.SetRunning(subservers.ACCOUNTS)
+		closeAccountService()
```

Comment:
Ah, I see. That's basically a fail on my part then, a `NewXXX` shouldn't do anything that requires stopping, only `Start()` should do that. Maybe add a short comment here then that explains the situation?

Reply:
Ok, agree that makes sense :)! Added a comment for the `closeAccountService` function.